clientapi/routing/membership.go

```diff
@@ -250,20 +255,10 @@ func sendInvite(
 	asAPI appserviceAPI.AppServiceInternalAPI, evTime time.Time,
 ) (util.JSONResponse, error) {
 	event, err := buildMembershipEvent(
-		ctx, userID, reason, profileAPI, device, "invite",
+		ctx, userID, reason, profileAPI, device, gomatrixserverlib.Invite,
 		roomID, false, cfg, evTime, rsAPI, asAPI,
 	)
-	if err == errMissingUserID {
-		return util.JSONResponse{
-			Code: http.StatusBadRequest,
-			JSON: jsonerror.BadJSON(err.Error()),
-		}, err
-	} else if err == eventutil.ErrRoomNoExists {
-		return util.JSONResponse{
-			Code: http.StatusNotFound,
-			JSON: jsonerror.NotFound(err.Error()),
-		}, err
```

Comment:
Removed because `errMissingUserID` is never returned and `checkMemberInRoom` already checks if the room exists and returns `IsInRoom=false`, resulting in an error returned earlier.

Reply:
`checkMemberInRoom` returns `IsInRoom=false`, either if the room doesn't exist or the user isn't in the room, that's what I mean.